imports/src/parser/parse_package.rs

```diff
@@ -82,25 +82,28 @@ impl ImportParser {
             .map_err(|error| ImportError::directory_error(error, &error_path, span))?;
 
         // Check if the imported package name is in the source directory.
-        let source_entries: Vec<DirEntry> = entries
-            .into_iter()
-            .filter(|entry| {
-                entry
-                    .file_name()
-                    .into_string()
-                    .unwrap()
-                    .trim_end_matches(SOURCE_FILE_EXTENSION)
-                    .eq(package_name)
-            })
-            .collect();
-
-        if source_entries.len() > 1 {
-            let paths: Vec<PathBuf> = source_entries.into_iter().map(|entry| entry.path()).collect();
-            return Err(ImportError::conflicting_local_imports(paths, span).into());
+        // let source_entries: Vec<DirEntry> = entries
+        let mut source_entries = entries.into_iter().filter(|entry| {
+            entry
+                .file_name()
+                .into_string()
+                .unwrap()
+                .trim_end_matches(SOURCE_FILE_EXTENSION)
+                .eq(package_name)
+        });
+
+        let matched_source_entry = source_entries.next();
+        if let Some(conflict_path) = source_entries.next() {
+            return Err(ImportError::conflicting_local_imports(
+                vec![
+                    matched_source_entry.map_or("".into(), |e| e.path()),
+                    conflict_path.path(),
+                ],
```

Comment:
The rest of `source_entries` never gets exhausted. You'll want `[matched_source_entry, conflict_path, ..source_entries]`, conceptually that is.

Reply:
I believe in this case we only need conflicting paths, so don't see the point in adding the rest.